components/gitpod-db/go/personal_access_token.go

```diff
@@ -87,6 +87,50 @@ func CreatePersonalAccessToken(ctx context.Context, conn *gorm.DB, req PersonalA
 	return token, nil
 }
 
+func UpdatePersonalAccessToken(ctx context.Context, conn *gorm.DB, tokenID uuid.UUID, userID uuid.UUID, hash string, expirationTime time.Time) (PersonalAccessToken, error) {
+	if tokenID == uuid.Nil {
+		return PersonalAccessToken{}, fmt.Errorf("Invalid or empty tokenID")
+	}
+	if userID == uuid.Nil {
+		return PersonalAccessToken{}, fmt.Errorf("Invalid or empty userID")
+	}
+	if hash == "" {
+		return PersonalAccessToken{}, fmt.Errorf("Token hash required")
+	}
+	if expirationTime.IsZero() {
+		return PersonalAccessToken{}, fmt.Errorf("Expiration time required")
+	}
+
+	db := conn.WithContext(ctx)
+
+	var token PersonalAccessToken
+	err := db.Transaction(func(tx *gorm.DB) error {
+		txErr := tx.
+			Where("id = ?", tokenID).
+			Where("userId = ?", userID).
+			Where("deleted = ?", 0).
+			Select("hash", "expirationTime").Updates(PersonalAccessToken{Hash: hash, ExpirationTime: expirationTime}).Error
+		if txErr != nil {
+			return txErr
+		}
+
+		txErr = tx.Where("id = ?", tokenID).Where("userId = ?", userID).Where("deleted = ?", 0).First(&token).Error
+		if txErr != nil {
+			return txErr
+		}
+
+		return nil
+	})
```

Comment:
I believe we can write this query using https://gorm.io/docs/update.html#Returning-Data-From-Modified-Rows

This would remove the need for 
* The explicit transaction
* The lookup afterwards

I'm just spinning up a workspace to test this out, will comment if I can make it work.

Reply:
Okay, looks like it doesn't work because it won't return the full `PersonalAccessToken` object, but only the rows which have actually changed. It's also dependant on the DB supporting Returning values.

Reply:
> Okay, looks like it doesn't work because it won't return the full PersonalAccessToken object, but only the rows which have actually changed. It's also dependant on the DB supporting Returning values.

Yep, we tried it (`clause.Returning{}`) yesterday, and it was not worked as expected, so we switched to update and select to avoid transaction

Reply:
> Okay, looks like it doesn't work because it won't return the full PersonalAccessToken object, but only the rows which have actually changed. It's also dependant on the DB supporting Returning values.

yep it's because mysql doesn't support `returning` sql clause